tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

```diff
@@ -182,8 +182,16 @@ export class LineChartComponent implements AfterViewInit, OnChanges, OnDestroy {
   }
 
   ngAfterViewInit() {
+    const dimPropChanged = this.readAndUpdateDomDimensions();
     this.initializeChart();
-    this.updateLineChart();
+    const viewboxPropChanged = this.updateLineChart();
+
+    // After view is initialized, if we ever change the Angular prop that should propagate
+    // to children, we need to retrigger the Angular change. Since we lazily update the
+    // property, these may return false.
+    if (dimPropChanged || viewboxPropChanged) {
+      this.changeDetector.detectChanges();
```

Comment:
When the component initializes but never resizes, I'm curious, why do we not need to call `this.lineChart.resize(this.domDimensions.main);` here, just like in `onViewResize` on L201?

Reply:
Ah, nvm I see now that `ChartImpl` calls its own `resize()` in its constructor, so LineChartComponent's `initializeChart` should take care of that.